test/structure/structure_region/test_capping.py

```diff
@@ -24,35 +25,184 @@ def test_capping_with_residue_terminals(helpers):
+
+
+
+
+def test_all_cap_types_unique_names():
+    """Checking that all the cap types do not conflict with any other atom names of other caps."""
+    test_stru = sp.get_structure(f"{DATA_DIR}tri_alanine.pdb")
+    sele = stru_sele.select_stru(
+        test_stru, "resi 2")
+
+    cap_types:List[str] = list(capping.SUPPORTED_CAPS.keys())
+
+    for nterm_cap in cap_types:
+        for cterm_cap in cap_types:
+            test_region = stru_regi.StructureRegion(atoms=sele.atoms)
+            capping.capping_with_residue_terminals(test_region, nterm_cap=nterm_cap, cterm_cap=cterm_cap)
+   
+            unique_names = set()
+            for atom in test_region.atoms:
+                unique_names.add(atom.name)
+        
+            assert len(test_region.atoms) == len(unique_names)
+
+
+
+def test_all_cap_types_have_realistic_bond_distances():
+    """Checking that all the cap types have realistic bond distances with their link Residue()'s and Atom()'s."""
+    test_stru = sp.get_structure(f"{DATA_DIR}tri_alanine.pdb")
+    sele = stru_sele.select_stru(
+        test_stru, "resi 2")
+
+    cap_types:List[str] = list(capping.SUPPORTED_CAPS.keys())
+
+    for nterm_cap in cap_types:
+        for cterm_cap in cap_types:
+            test_region = stru_regi.StructureRegion(atoms=sele.atoms)
+            capping.capping_with_residue_terminals(test_region, nterm_cap=nterm_cap, cterm_cap=cterm_cap)
+         
+            for atom in test_region.atoms:
+                if not isinstance(atom.parent, capping.ResidueCap):
+                    continue
+                link_distance:float=atom.parent.link_atom.distance_to(atom.parent.socket_atom)
+                assert link_distance >= 1.0 and link_distance <= 1.50
+
+
+
+def test_correct_num_caps():
+    """Testing that the correct number of caps are made."""
+
+    test_stru = sp.get_structure(f"{DATA_DIR}tri_alanine.pdb")
+    sele = stru_sele.select_stru(
+        test_stru, "resi 1+2+3")
+
+    test_region = stru_regi.StructureRegion(atoms=sele.atoms)
+
+    caps = set()
+
+    capping.capping_with_residue_terminals(test_region)
+
+    for atom in test_region.atoms:
+        if not isinstance(atom.parent, capping.ResidueCap):
+            continue
+        caps.add(atom.parent)
+
+    assert len(caps) == 0
+
+
+    sele = stru_sele.select_stru(
+        test_stru, "resi 1")
+
+    test_region = stru_regi.StructureRegion(atoms=sele.atoms)
+
+    caps = set()
+
+    capping.capping_with_residue_terminals(test_region)
+
+    for atom in test_region.atoms:
+        if not isinstance(atom.parent, capping.ResidueCap):
+            continue
+        caps.add(atom.parent)
+
+    assert len(caps) == 1 
+
+
+    sele = stru_sele.select_stru(
+        test_stru, "resi 3")
+
+    test_region = stru_regi.StructureRegion(atoms=sele.atoms)
+
+    caps = set()
+
     capping.capping_with_residue_terminals(test_region)
-    # test
-    atoms = test_region.atoms
```

Comment:
Please also make this test work. I can help if you need!

Reply:
Sure I can ig

Reply:
This has been fixed! still works perfectly :)